service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/AddressTagService.kt

```diff
@@ -0,0 +1,57 @@
+package io.provenance.digitalcurrency.consortium.service
+
+import com.google.protobuf.ByteString
+import io.provenance.digitalcurrency.consortium.config.BankClientProperties
+import io.provenance.digitalcurrency.consortium.config.logger
+import io.provenance.digitalcurrency.consortium.domain.AddressRegistrationRecord
+import io.provenance.digitalcurrency.consortium.domain.AddressRegistrationStatus
+import io.provenance.digitalcurrency.consortium.extension.toByteArray
+import org.springframework.stereotype.Service
+
+@Service
+class AddressTagService(
+    private val pbcService: PbcService,
+    private val bankClientProperties: BankClientProperties
+) {
+    private val log = logger()
+
+    fun createEvent(addressRegistrationRecord: AddressRegistrationRecord) {
+        val existing =
+            pbcService.getAttributeByTagName(addressRegistrationRecord.address, bankClientProperties.kycTagName)
```

Comment:
I'm not checking that the value in the tag matches the bank account uuid and also not checking for more than one tag with the same name existing because we assume one registration per address...but I still feel like I should be checking the value? Thoughts?

Reply:
i feel like if we get a request to register an address that we already have a record for we 400 so that can't happen for now?

Reply:
yes we send a 400 if they try to either register an existing bank account uuid or an existing address